llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/docs/base.py

```diff
@@ -43,8 +43,8 @@ def load_data(
         fs: Optional[AbstractFileSystem] = None,
     ) -> List[Document]:
         """Parse file."""
-        if not isinstance(file, Path):
-            file = Path(file)
+        if not isinstance(file, Path) and not isinstance(file, PurePosixPath):
+            file = PurePosixPath(file)
```

Comment:
I think we should only be using `PurePosixPath` when `fs` is some remote file system -- and probably only in simple directory reader?